FusionIIIT/applications/online_cms/views.py

```diff
@@ -1,3 +1,165 @@
-# from django.shortcuts import render
+from __future__ import unicode_literals
 
-# Create your views here.
+import os
+import subprocess
+from datetime import datetime
+
+from django.conf import settings
+from django.contrib.auth.decorators import login_required
+from django.core.files.storage import FileSystemStorage
+from django.http import HttpResponse
+from django.shortcuts import render
+
+from applications.academic_information.models import (Course, Instructor,
+                                                      Student)
+from applications.academic_procedures.models import Register
+from applications.globals.models import ExtraInfo
+
+from .forms import AddDocuments, AddVideos
+from .helpers import semester
+from .models import CourseDocuments, CourseVideo
+
+
+@login_required
+def viewcourses(request):
+    user = request.user
+    extrainfo = ExtraInfo.objects.get(user=user)
+    if extrainfo.user_type == 'student':
+        student = Student.objects.get(id=extrainfo)
+
+        roll = student.id.id[:4]
+        register = Register.objects.filter(student_id=student, semester=semester(roll))
+        return render(request, 'online_cms/viewcourses.html',
+                      {'register': register,
+                       'extrainfo': extrainfo})
+    else:
+        instructor = Instructor.objects.filter(instructor_id=extrainfo)
+        return render(request, 'online_cms/viewcourses.html',
+                      {'instructor': instructor,
+                       'extrainfo': extrainfo})
+
+
+@login_required
+def course(request, course_code):
+    user = request.user
+    extrainfo = ExtraInfo.objects.get(user=user)
+
+    if extrainfo.user_type == 'student':
+        student = Student.objects.get(id=extrainfo)
+        roll = student.id.id[:4]
+        course = Course.objects.filter(course_id=course_code, sem=semester(roll))
+        instructor = Instructor.objects.get(course_id=course)
```

Comment:
This will give error because `course_id` will expect a `course instance` but you are giving it a query_set, `(using filter)`